scip_indexer/SCIPIndexer.cc

```diff
@@ -173,6 +187,15 @@ class SCIPState {
     UnorderedSet<pair<core::FileRef, string>> emittedSymbols;
     UnorderedMap<core::FileRef, vector<scip::SymbolInformation>> symbolMap;
 
+    /// Stores the relationships that apply to a field or a method.
+    ///
+    /// Ideally, 'UntypedGenericSymbolRef' would not be duplicated across files
+    /// but we keep these separate per file to avoid weirdness where logically
+    /// different but identically named classes exist in different files.
```

Comment:
Now that I'm thinking about this again, I'm a little sus on this reasoning. I should double-check what RubyMine does here. In addition, it is unconsistent that FieldResolver carries a cache which doesn't have a `FileRef` as part of the key. In the normal case, that is what you would want, I think, because if you are extending a class with methods in a different file, you more likely mean the same class rather than a different identically named class.

Reply:
OK, RubyMine does connect classes even if they're in different files, but if they are identically named. However, on doing Go to Def, it prefers navigating within the same file instead of showing results in other files.

Somewhat counterintuitively, RubyMine also connects stuff across different "levels". For example, if you have a class `C` and another inside a module `M.C`, then Cmd+Click on a def `M.C.@f` will also show the `C.@f`. So that part definitely seems search-based...

Reply:
Resolved this by using an extra FileRef in FieldResolver.